src/terrain/terrain_base.cpp

```diff
@@ -39,13 +39,13 @@ TerrainBase::qb_read(
 }
 
 TerrainBase::TerrainBase(
-    int x, int y, int Area_size, int z,
-    const std::map<uint8_t, const Material>& materials, const Json::Value& biome_data,
+    Dim x, Dim y, Dim Area_size, Dim z,
```

Comment:
```suggestion
    Dim x, Dim y, Dim z, Dim area_size, 
```

Please put these in a sensible order